[WIP] Clean up index README.md, including doc ToC, and merge architecture and northbound documentations. (#1470)
telefonicaid/iotagent-node-lib: doc/api.md

        -   [Triggering commands](#triggering-commands)
        -   [Command reception](#command-reception)
        -   [Command confirmation](#command-confirmation)

>     -   [TimeInstant and Timestamp flag](#timeinstant-and-timestamp-flag)
>         -   [Timestamp Compression](#timestamp-compression)

Author: ```suggestion
    -   [TimeInstant and Timestamp flag](#timeinstant-and-timestamp-flag)
```

    is optional. They are called internally as _active attributes_.

-   **`lazy`**: Passive measures that are pulled from the device to the IoT agent. When a request for data from a lazy
    attribute arrives to the Context Broker, it forwards the request to the IoT Agent (that behaves as NGSI Context
    Provider for all the lazy attributes or commands). The IoT Agent will then ask the device for the information
    needed, transform that information to a NGSI format and return it to the Context Broker. This operation will be
    synchronous from the customer perspective: the Context Broker won't return a response until the device has returned
    its response to the IoT Agent. For each attribute, its `name` and `type` must be provided. They are called
    internally as _lazy attributes_.

-   **`static`**: It is static attributes that are persisted in the Context Broker. They are not updated by the device,
    but they can be modified by the user. They are useful to store information about the device that is not updated by
    the device itself. For instance, a `location` static attribute is can be used to store the location of a fixed
    device.

> -   **`commands`**: Commands are actions that can be invoked in the device. They are entity attributes, but they are not

Reviewer: Maybe a link to the recently added section about Command Execution would be useful at this point.

Author: Added here: https://github.com/telefonicaid/iotagent-node-lib/pull/1470/commits/fe57c79fedcca7b44ffb7fa7b2e26a7df0fcd402

}
```

## TimeInstant and Timestamp flag

As part of the device to entity mapping process, the IoT Agent creates and updates automatically a special timestamp
attribute called `TimeInstant`. This timestamp is represented as two different properties of the mapped entity:

> -   In the case of NGSI-v2, a `TimeInstant` **metadata** is added in each updated attribute. With NGSI-LD, the Standard
>     `observedAt` property-of-a-property is used instead.

Author: ```suggestion
-   An attribute `TimeInstant` is added to updated entities in the case of NGSI-v2, which captures as an ISO8601
    timestamp when the associated measurement was observed. With NGSI-LD, the Standard
    `observedAt` property is used instead
```

}
```

## TimeInstant and Timestamp flag

As part of the device to entity mapping process, the IoT Agent creates and updates automatically a special timestamp
attribute called `TimeInstant`. This timestamp is represented as two different properties of the mapped entity:

-   An attribute `TimeInstant` is added to updated entities in the case of NGSI-v2, which captures as an ISO8601
    timestamp when the associated measurement was observed. With NGSI-LD, the Standard

>     `observedAt` property is used instead
> 

Author: ```suggestion
    `observedAt` property is used instead
```

-   If the resulting `TimeInstant` not follows [ISO_8601](https://en.wikipedia.org/wiki/ISO_8601) (either from a direct
    measure of after a mapping, as described in the previous bullet) then it is refused (so a failover to server
    timestamp will take place).

>                          
> -   the timestamp of different attributes belonging to the same measurement record may not be equal.
> -   the arrival time and the measurement timestamp will not be the same in the general case (when explicitly defining
>     the timestamp in the measurement)
> -   if `timezone` field is defined as part of the provisioning of the device or group, timestamp fields will be
>     generated using it. For instance, if `timezone` is set to `America/Los_Angeles`, a possible timestamp could be

Author: ```suggestion
-   the timestamp of different attributes belonging to the same measurement record may not be equal.
-   the arrival time and the measurement timestamp will not be the same in the general case (when explicitly defining
    the timestamp in the measurement)
-   if `timezone` field is defined as part of the provisioning of the device or group, timestamp fields will be
    generated using it. For instance, if `timezone` is set to `America/Los_Angeles`, a possible timestamp could be
```
